### PR TITLE
create index with dynamically allocated blockmaps

### PR DESCRIPTION
**Pull request by Ngalstyan4**

### CODE REVIEW

**Review by Ngalstyan4 — COMMENTED** [review]
I think this is ready to be merged. @davkhech @var77 , please take a look before I squash the commits and merge!

File: Dockerfile.dev [hunk 1]
```diff
@@ -5,7 +5,8 @@ COPY . .
 # Set Locale
 RUN apt update && apt-mark hold locales && \
 # Install required packages for build
-apt install -y --no-install-recommends build-essential cmake postgresql-server-dev-15 gdb
+apt install -y --no-install-recommends build-essential cmake postgresql-server-dev-15 gdb wget postgres-15-pgvector python3-pip sudo && \
```
[the remark this answers is not in the document]
Comment (varik-coinstats) [reply Dockerfile.dev:8]: Yes they’re necessary for the livedebug script to work in the container

[the remark this answers is not in the document]
Comment (var77) [reply Dockerfile.dev:8]: Yes they're necessary for the livedebug script to work in the container.

File: src/hnsw/insert.c [hunk 2]
```diff
@@ -98,6 +98,7 @@ bool ldb_aminsert(Relation         index,
         hdr_page = BufferGetPage(hdr_buf);
         hdr = (HnswIndexHeaderPage *)PageGetContents(hdr_page);
         assert(hdr->magicNumber == LDB_WAL_MAGIC_NUMBER);
+        HEADER_FOR_EXTERNAL_RETRIEVER = *hdr;
```
Comment (davkhech) [inline src/hnsw/insert.c:101]: no need to add here, as it's assigned later out of if block

**Comment by davkhech** [comment]
LGTM!

File: src/hnsw/insert.c (hunk 2 above)
Comment (Ngalstyan4) [reply src/hnsw/insert.c:101]: needed because `ldb_wal_retriever_area_init` uses it :)

**Pull request merged**